server/aesdsocket.c

```diff
@@ -0,0 +1,253 @@
+	hints.ai_flags = AI_PASSIVE;
+	
+	state = getaddrinfo(NULL, PORT, &hints, &param);
+	if (state != 0)
+	{
+		syslog(LOG_ERR, "Error !! getaddrinfo failed \n");
+		exit(EXIT_FAILURE);
+	}
+	
+	socktfd = socket(AF_INET6, SOCK_STREAM, 0);
+	if (socktfd == -1)
+	{
+		syslog(LOG_ERR, "Error !! Socket connection failed \n");
+		exit(EXIT_FAILURE);
+	}
+	
+	state = bind(socktfd, param->ai_addr, param->ai_addrlen);
+	if (state == -1)
+	{
+		syslog(LOG_ERR, "Error !! Binding failed : %d\n", state);
+		exit(EXIT_FAILURE);
+	}
+
+	freeaddrinfo(param);
+	
+	int file_fd;
+	file_fd = creat(PATH, 0666);
+	if (file_fd == -1)
+	{
+		syslog(LOG_ERR, "Error !! File creation failed : %d\n", file_fd);
+		exit(EXIT_FAILURE);
+	}
+	
+	close(file_fd);
+	
+	bool p_state = true;
+	ssize_t data_rx = 0;
+	ssize_t data_wr = 0;
+	
+	while(1)
+	{
+		client_buff = (char*) malloc((sizeof(char) *BUFFER_SIZE));
+		if (client_buff == NULL)
+		{
+			syslog(LOG_ERR, "Failed to allocate memory\n");
+			exit(EXIT_FAILURE);
+		}
+
+		memset(client_buff, 0, BUFFER_SIZE);
+
+		state = listen(socktfd, 10);
+		if (state == -1)
+		{
+			syslog(LOG_ERR, "Error !! Listen failed:%d\n", state);
+			exit(EXIT_FAILURE);
+		}
+
+		client_addr_len = sizeof(struct sockaddr);
+
+		clientfd = accept(socktfd, (struct sockaddr *) &client_addr, &client_addr_len);
+		if (clientfd == -1)
+		{
+			syslog(LOG_ERR, " Error !! Accepting failed\n");
+			exit(EXIT_FAILURE);
+		}
+		
+		p_state = false;
+		
+		
+		while (!p_state)
+		{
+			
+			data_rx = recv(clientfd, buff_rx, BUFFER_SIZE, 0);
+			if (data_rx == 0)
+			{
+				syslog(LOG_INFO,"Reception success\n");
+			}
+			else if (data_rx < 0)
+			{
+				syslog(LOG_ERR,"Error !! Recieve failed:%d\n", state);
+				exit(EXIT_FAILURE);
+			}
+			
+			int i;
+			
+			for (i = 0; i < BUFFER_SIZE; i++)
+			{
+				if (buff_rx[i] == '\n')
+				{
+					p_state = true;
+					i++;
+					break;
+				}
+			}
+
+			len += i;
+
+			client_buff = (char*) realloc(client_buff, (len + 1));
+			if (client_buff == NULL)
+			{
+				syslog(LOG_ERR,"Error !! Realloc failed\n");
+				exit(EXIT_FAILURE);
+			}
+			strncat(client_buff, buff_rx, i);
+			memset(buff_rx, 0, BUFFER_SIZE);
+		}
+		
+		
+		file_fd = open(PATH, O_APPEND | O_WRONLY);
+		if (file_fd == -1)
+		{
+			syslog(LOG_ERR,"Error !! Failed to open file\n");
+			exit(EXIT_FAILURE);
+		}
+		
+		data_wr = write(file_fd, client_buff, strlen(client_buff));
+		if (data_wr == -1)
+		{
+			syslog(LOG_ERR,"Error !! Write failed \n");
+			exit(EXIT_FAILURE);
+		}
+		else if (data_wr != strlen(client_buff))
+		{
+			syslog(LOG_ERR,"Partial Write\n");
+			exit(EXIT_FAILURE);
+		}
+		else
+		{
+			syslog(LOG_INFO,"File write success!\n");
+		}
+
+		close(file_fd);
+		
+	
+		memset(buff_rx, 0, BUFFER_SIZE);
+		file_fd = open(PATH, O_RDONLY);
+		if (file_fd == -1)
+		{
+			syslog(LOG_ERR,"Error !! Open Failed to read\n");
+			exit(EXIT_FAILURE);
+		}
+		
+		char char_rd = 0;
+		
+		for (int i = 0; i < len; i++)
```

Comment:
@Ruthvik-Chanda what will happen here when the condition described above with `data_rx` is < `BUFFER_SIZE` and `buff_rx` not containing a `\n`?